kong/llm/drivers/shared.lua

```diff
@@ -143,6 +179,78 @@ local function handle_stream_event(event_table, model_info, route_type)
   end
 end
 
+---
+-- Splits up a string by delimiter, whilst preserving
+-- empty lines, double line-breaks.
+--
+-- @param {string} str input string to split
+-- @param {string} delimiter delimeter (can be complex string) to split by
+-- @return {table} n number of split results, or empty table
+local function complex_split(str, delimiter)
```

Comment:
This function is not needed.  Use `split` from kong.tools.utils.

Reply:
Yep you're right the kong PL split works. I swore I tested this before and it broke, but I've changed it now.